stdlib/LinearAlgebra/src/special.jl

```diff
@@ -3,6 +3,12 @@
 # Methods operating on different special matrix types
 
 
+# Usually, reducedim_initarray calls similar, which yields a sparse matrix for a
+# Diagonal/Bidiagonal/Tridiagonal/SymTridiagonal matrix. However, reducedim should
+# yield a dense vector to increase performance.
+Base.reducedim_initarray(A::Union{Diagonal,Bidiagonal,Tridiagonal,SymTridiagonal}, region, init, ::Type{R}) where {R} = fill(convert(R, init), Base.reduced_indices(A,region))
```

Comment:
Now I'm confused. None of your new `_sum` methods calls this 4-arg method...

Reply:
The method in https://github.com/JuliaLang/julia/blob/8617860b2cf3604b1ff365fa5f3f5d00f3ca47cc/base/reducedim.jl#L93 redirects the calls in my new `_sum` methods.